File: tests/perf/test-call-basic.rb
def test()
    def f() return end

    i = 0
    while i < 1e7 do
        f()
        f()
        f()
        f()
        f()
        f()
        f()
        f()
        f()
        f()
        i += 1
    end
end

test()

File: tests/perf/test-call-basic.js
/*
 *  Basic function call performance.
 */

function test() {
    var i;

    function f() { return; }

    for (i = 0; i < 1e7; i++) {
        f();
        f();
        f();
        f();
        f();
        f();
        f();
        f();
        f();
        f();
    }
}

try {
    test();
} catch (e) {
    print(e.stack || e);
    throw e;
}
